evaluationquestions.txt

```diff
@@ -0,0 +1,91 @@
+////////////////    part A      ////////////////
+initialize an array with 2 values in it (apples and oranges)
+const fruit = ["apples", "oranges"];
+
+what does array.map do? and what is returns?
+map.array method creates a new array by iterating over the array and calling function on every element of array, it return the same array length with the ability to change array elements
+fruit.map(element => {
+    element * 3.5;
+});
+
+what does array.filter do? and what is returns?
+array.filter method creates an array filter filled with the array filtered elements, it return the filtered element of the array with the ability to change these returned elements
+fruit.filter(element => {
+    element.status === 'intern';
+}).filter(age => {
+    age.age > 23;
+});
+
+what does array.reduce do? and what is returns?
+array.reduce method used to change the array into a single value, it ilterates over the whole array doining changes on each value from left-to-right 
+fruit = fruit.reduce(sum,element => {
+    sum + element.age;
+}, 0);
+
+////////////////    part B      ////////////////
+what is the difference between Synchronous and ASynchronous?
+Synchronous calls refer to a callback where the code execution waits for an event before continuing, while Asynchronous calls, on the other hand, refer to a callback that does not block the execution of the program.
+
+what are promises ?
+Promis is program commiting to do something if resolved, it represent processes that are already happening, which can be chained with callback functions.
```

Comment:
what do you mean program?

Reply:
promis is when a program commiting to do something, forget to put when.

Reply:
done